app/assets/javascripts/autocise_textarea.js

```diff
@@ -0,0 +1,53 @@
+// Get the max height and apply
+// to all textareas when page loads/reloads
+
+$(document).ready(function() {
+  var numberArray = [];
+  $('textarea').each(function(textarea) {
+    $(this).height( $(this)[0].scrollHeight );
+    numberArray.push($(this).innerHeight());
+    var max = Math.max(...numberArray);
+    $(this).innerHeight(max);
+  });
+});
+
+// Use an event to listen to all input events and then
+// filter out ones that are not on a textarea
+// if the element is a textarea we will call a
+// autoExpandTextarea function and pass in
+// the element as an argument using event.target.
+document.addEventListener('input', function(event) {
+  if (event.target.tagName.toLowerCase() !== 'textarea') return;
+  autoExpandTextarea(event.target);
+}, false);
+
+var autoExpandTextarea = function(field) {
+  // check the field parameter class so we can give the
+  // same height to other textareas with same class.
+  var isMain = field.classList.contains('item-name');
+  var isMaori = field.classList.contains('item_maori_name');
+
+  // Reset field height
+  field.style.height = 'inherit';
+
+  // get computed styles for the element and
+  // scrollheight to calculate height of the content.
+  var computed = window.getComputedStyle(field);
+
+  // Calculate the height
+  var height = parseInt(computed.getPropertyValue('border-top-width'), 10)
+               + parseInt(computed.getPropertyValue('padding-top'), 10)
+               + field.scrollHeight
+               + parseInt(computed.getPropertyValue('padding-bottom'), 10)
+               + parseInt(computed.getPropertyValue('border-bottom-width'), 10);
+  // Reset height of element using style property or just use the ifs?
+  field.style.height = height + 'px';
+
+  // alter textarea row height to make consistent and even.
+  if(isMain || isMaori) {
+    for(var i = 0, j = 0; i < isMain.length, j < isMaori.length; i++, j++) {
```

Comment:
If you're using the same value for `i` and `j`, I would recommend just using one. It's also best practice to give a more descriptive name to your iterators, maybe `sign`?

Reply:
mmmm

Reply:
yes right needs fracturing. The I and j act as indexes.